comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -91,6 +92,15 @@ func (fh *forwarderHealth) init() {
 	if apiKeyCount != 0 {
 		fh.timeout /= time.Duration(apiKeyCount)
 	}
+
+	// when the config updates the "api_key", process that change
+	if fh.config != nil {
+		fh.config.OnUpdate(func(key string) {
```

Comment:
Not directly related to this PR: What do you think about having the following API for notifying when an configuration option is updated?

```go
config.OnStringUpdate("api_key", func (new_key string) { // func is called only when "api_key" changed
    fh.updateAPIKey(new_key)
} 
```
which is equivalent to
```go
config.OnStringUpdate("api_key", fh.updateAPIKey)

```

Reply:
I think that's a good idea. After using `OnUpdate` a few times I've realized it's always necessary to check the key to see if the expected key is being updated, and right now it's very easy to mess that up and forget to do it.